Camacho, C., Coulouris, G., Avagyan, V., Ma, N., Papadopoulos, J., Bealer, K., & Madden, T. L. (2009). BLAST+: architecture and applications.. BMC Bioinformatics, 10(1), 421-421.
Ebersberger, I., Strauss, S., & Haeseler, A. V. (2009). Hamstr: profile hidden markov model based search for orthologs in ests. BMC Evolutionary Biology, 9(1), 1-9.
Edgar, R. C. (2004). Muscle: multiple sequence alignment with high accuracy and high throughput. Nucleic Acids Research, 32(5), 1792-1797.
Gouy, M., Guindon, S., & Gascuel, O. (2010). SeaView Version 4: A Multiplatform Graphical User Interface for Sequence Alignment and Phylogenetic Tree Building. Molecular Biology and Evolution, 27(2), 221-224.
Grabherr, M., Haas, B. J., Yassour, M., Levin, J. Z., Thompson, D. A., Amit, I., ... & Regev, A. (2011). Full-length transcriptome assembly from RNA-Seq data without a reference genome. Nature Biotechnology, 29(7), 644-652.
Hackett, S. J., Kimball, R. T., Reddy, S., Bowie, R. C. K., Braun, E. L., Braun, M. J., Chojnowski, J. L., Cox, W. A., Han, K.-L., Harshman, J., Huddleston, C. J., Marks, B. D., Miglia, K. J., Moore, W. S., Sheldon, F. H., Steadman, D. W., Witt, C. C. & Yuri, T. (2008). A phylogenomic study of birds reveals their evolutionary history. Science, 320, 1763– 1768.
Jarvis, E. D., Mirarab, S., Aberer, A. J., Li, B., Houde, P., & Li, C., et al. (2014). Whole-genome analyses resolve early branches in the tree of life of modern birds. Science, 346(6215), 1320.
Jarvis, E. D., Mirarab, S., Aberer, A. J., Li, B., Houde, P., & Li, C., et al. (2015). Phylogenomic analyses data of the avian phylogenomics project. GigaScience, 4(1), 4.
Jeffroy, O., Brinkmann, H., Delsuc, F., & Philippe, H. (2006). Phylogenomics : the beginning of incongruence?. Trends in Genetics, 22(4), 225-231.
Katoh, K., & Standley, D. M. (2013). Mafft multiple sequence alignment software version 7: improvements in performance and usability. Molecular Biology & Evolution, 30(4), 772-80.
Kriventseva, E. V., Rahman, N., Espinosa, O., & Zdobnov, E. M. (2008). OrthoDB: the hierarchical catalog of eukaryotic orthologs. Nucleic Acids Research, 36, 271-275.
Kück, P. (2009) ALICUT: a Perlscript which cuts ALISCORE identified RSS. Department of Bioinformatics, Zoologisches Forschungsmuseum A. Koenig (ZFMK), Bonn, Germany, version 2.0 edition.
Lepage, D., Warnier, J. (2014). The Peters' Check-list of the Birds of the World (1931-1987) Database. Accessed on 12/06/2017 from Avibase, the World Database: http://avibase.bsc-eoc.org/peterschecklist.jsp.
Maddison, D. R., Swofford, D. L., & Maddison, W. P. (1997). Nexus: an extensible file format for systematic information. Systematic Biology, 46(4), 590-621.
Mayr, G. (2011). Metaves, Mirandornithes, Strisores and other novelties – a critical review of the higher-level phylogeny of neornithine birds. Journal of Zoological Systematics and Evolution- ary Research, 49, 58–76.
McCormack, J. E., Harvey, M. G., Faircloth, B. C., Crawford, N. G., Glenn, T. C. & Brumfield, R. T. (2013). A phylogeny of birds based on over 1,500 loci collected by target enrichment and high-throughput sequencing. PLoS ONE, 8, e54848.
Misof, B., & Misof, K. (2009). A monte carlo approach successfully identifies randomness in multiple sequence alignments: a more objective means of data exclusion. Systematic Biology, 58(1), 21-34.
Mistry, J., Finn, R. D., Eddy, S. R., Bateman, A., & Punta, M. (2013). Challenges in homology search: HMMER3 and convergent evolution of coiled-coil regions. Nucleic Acids Research, 41(12).
Prum, R. O., Berv, J. S., Dornburg, A., Field, D. J., Townsend, J. P., & Lemmon, E. M., et al. (2015). A comprehensive phylogeny of birds (aves) using targeted next-generation dna sequencing. Nature, 526(7574), 569.
Sangster, G. (2005). A name for the flamingo–grebe clade. Ibis, 147, 612–615.
Suh, A., Paus, M., Kiefmann, M., Churakov, G., Franke, F. A., Brosius, J., Kriegs, J. O. & Schmitz, J. (2011). Mesozoic retro- posons reveal parrots as the closest living relatives of passerine birds. Nature Communications, 2, 443.
Suh, A., Smeds, L., & Ellegren, H. (2015). The dynamics of incomplete lineage sorting across the ancient adaptive radiation of neoavian birds. PLoS biology, 13(8), e1002224.
Swofford, D. (2002). Paup: phylogenetic analysis using parsimony (*and other methods) sunderland. Mccarthy.
Wu, H., Wang, Y., Xie, Q., Ke, Y., & Bu, W. (2016). Molecular classification based on apomorphic amino acids (Arthropoda, Hexapoda): Integrative taxonomy in the era of phylogenomics. Scientific Reports 6: 28308.
Yuri, T., Kimball, R. T., Harshman, J., Bowie, R. C. K., Braun, M. J., Chojnowski, J. L., Han, K.-L., Hackett, S. J., Huddleston, C. J., Moore, W. S., Reddy, S., Sheldon, F. H., Steadman, D. W., Witt, C. C. & Braun, E. L. (2013). Parsimony and model- based analyses of indels in avian nuclear genes reveal congruent and incongruent phylogenetic signals. Biology, 2, 419–444.
Zdobnov, E. M., Tegenfeldt, F., Kuznetsov, D., Waterhouse, R. M., Simão, F. A., & Ioannidis, P., et al. (2016). Orthodb v9.1: cataloging evolutionary and functional annotations for animal, fungal, plant, archaeal, bacterial and viral orthologs. Nucleic Acids Research, 45(Database issue), D744-D749.
Zhang, G., Li, B., Li, C., Gilbert, M. T., Jarvis, E. D., & Wang, J. (2014). Comparative genomic data of the avian phylogenomics project. GigaScience, 3(1), 26.
